src/app/shared/components/addregressionregion/addregressionregion.component.html

```diff
@@ -0,0 +1,102 @@
+          <button class="close pull-right" (click)="d('closed')">x</button>
+      </div>
+  </div>
+  <div class="modal-body">
+      <h3 class="modal-title" *ngIf="addRegReg">New Regression Region</h3>
+      <h3 class="modal-title" *ngIf="!addRegReg">Edit Regression Region</h3>
+      <form [formGroup]="newRegRegForm" style="margin-bottom: 10px;">
+          <!-- Regression Type Name* -->
+          <div class="form-group required" [ngClass]="{'form-invalid': !newRegRegForm.get('name').valid && newRegRegForm.get('name').dirty}">
+              <label class="req" for="name">Name:</label>
+              <input class="form-control" type="text" formControlName="name" required>
+              <div class="input-invalid-warning" *ngIf="!newRegRegForm.get('name').valid && newRegRegForm.get('name').dirty">Regression
+                  Name is required</div>
+          </div>
+
+          <!-- Description -->
+          <div class="form-group">
+              <label for="description">Description:</label>
+              <textarea class="form-control" rows="3" placeholder="" formControlName="description"></textarea>
+          </div>
+
+          <!-- Short Name -->
+          <div class="form-group required" [ngClass]="{'form-invalid': !newRegRegForm.get('code').valid && newRegRegForm.get('code').dirty || (newRegRegForm.get('code').value != null && !newRegRegForm.get('code').value.startsWith('GC'))}">
+              <label class="req" for="code">Code (e.g. GC1234):</label>
+              <input class="form-control" type="text" formControlName="code">
+              <div class="input-invalid-warning" *ngIf="!newRegRegForm.get('code').valid && newRegRegForm.get('code').dirty && newRegRegForm.get('code').value == ''">Code
+                  is required</div>
+              <div class="input-invalid-warning" *ngIf="newRegRegForm.get('code').value != '' && newRegRegForm.get('code').value != null && !newRegRegForm.get('code').value.startsWith('GC')">
+                      Code must start with "GC"</div>
+          </div>
+          <!-- Region* -->
+          <div class="form-group required" [ngClass]="{ 'form-invalid': !newRegRegForm.get('state').valid && newRegRegForm.get('state').dirty }">
+              <label class="req" for="state">Region:</label>
+              <select class="form-control" formControlName="state">
+                  <option *ngFor="let reg of regions" [value]="reg.id">{{ reg.name }}</option>
+              </select>
+              <div class="input-invalid-warning" *ngIf="!newRegRegForm.get('state').valid && newRegRegForm.get('state').dirty">
+                  Region is required
+              </div>
+          </div>
+          
+          <button *ngIf="!uploadPolygon" type="button" class="btn-blue" (click)="showMap()">Upload Polygon</button>
+          <button *ngIf="!addCitation" type="button" class="btn-blue" (click)="addCitation = true" style="margin-left: 5px;">Add Citation</button>
```

Comment:
My one comment here is that the flow is a little bit weird here, and I think moving those buttons would help. As it is now, the "Upload Polygon" button is above the citation form and before the "Add Citation" button if there's no citation. If you click it, the map shows up at the bottom. I think if you move these buttons to between the html for the new citation form and the map, with the upload polygon button after the add citation button, then the buttons would still be in line with each other but then the forms would show up closer to where the buttons are. Hope that makes sense.

Here's a recordit to show the workflow: https://recordit.co/PabQH96W6f